### PR TITLE
Httat 8 material color theme

### PR DESCRIPTION
**Pull request by katiehom**
# Description
Added sass folder with custom material color theme and applied to existing components to test.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [ ] Unit test

# Checklist:
- [x] I have pulled code from main and re-tested changes before opening the PR
- [x] No new bugs or vulnerabilities have been found in the code
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by katiehom** [comment]
We will likely want to delete `_color-palletes.scss` and `_index.scss` eventually as they won't be needed. We tried to incorporate a `secondary` color theme but were unable to incorporate it into the current theme as it only allowed for "primary", "accent", and "warn". We tried doing a merge but that was not successful.

**Comment by meghannfh** [comment]
[HTTAT-8 - Material Color Scheme](https://trello.com/c/jxz1iYyD/8-httat-8-material-color-scheme)

**Comment by GUIMess** [comment]
Did we lose the list of grades? 
image

**Comment by bob-fornal** [comment]
> Did we lose the list of grades?

Grades are working on my machine. You might have tested before main was merged in.

**Comment by bob-fornal** [comment]
> We will likely want to delete `_color-palletes.scss` and `_index.scss` eventually as they won't be needed. We tried to incorporate a `secondary` color theme but were unable to incorporate it into the current theme as it only allowed for "primary", "accent", and "warn". We tried doing a sass theme merge but that was not successful.

No problem.

File: frontend/src/app/app.component.scss
Comment (bob-fornal) [inline frontend/src/app/app.component.scss:1]: I'm pretty sure this file was removed, since it's empty.

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { Component } from '@angular/core';
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
```
Comment (bob-fornal) [inline frontend/src/app/app.component.ts:6]: This is the line that can be removed once the `app.component.scss` files is deleted.

File: frontend/src/app/features/assistance-selection/assistance-selection.component.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <button
       class="assistance-level-button"
       mat-raised-button
-      color="primary"
+      color="accent"
```
Comment (bob-fornal) [inline frontend/src/app/features/assistance-selection/assistance-selection.component.html:6]: Did we want to leave this `accent`, or revert back to `primary`?

File: frontend/src/app/features/grade-level-selection/grade-level-selection.component.scss [hunk 3]
```diff
@@ -9,7 +9,7 @@
     margin-left: 0.25em;
   }
   &.selected {
-    background-color: green;
+    background-color: httat-theme(primary); //$blue;
```
Comment (bob-fornal) [inline frontend/src/app/features/grade-level-selection/grade-level-selection.component.scss:12]: I thought we were going with `var(--override--selected-bg-color)` here and on the assistance selection component?

File: frontend/src/app/features/grade-level-selection/grade-level-selection.component.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import config from '../../core/constants/config.json';
   // eslint-disable-next-line @angular-eslint/component-selector
   selector: 'grade-level-selection',
   templateUrl: './grade-level-selection.component.html',
-  styleUrls: ['./grade-level-selection.component.scss'],
+  styleUrls: ['./grade-level-selection.component.scss']
```
Comment (bob-fornal) [inline frontend/src/app/features/grade-level-selection/grade-level-selection.component.ts:10]: There should be a comma here per ESLint.

File: frontend/src/app/features/header/header.component.ts [hunk 5]
```diff
@@ -4,6 +4,9 @@ import { Component } from '@angular/core';
   // eslint-disable-next-line @angular-eslint/component-selector
   selector: 'header',
   templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss'],
+  styleUrls: ['./header.component.scss']
```
Comment (bob-fornal) [inline frontend/src/app/features/header/header.component.ts:7]: Comma here per ESLint

File: frontend/src/app/features/header/header.component.ts [hunk 6]
```diff
@@ -4,6 +4,9 @@ import { Component } from '@angular/core';
...
 })
-export class HeaderComponent {}
+export class HeaderComponent {
```
Comment (bob-fornal) [inline frontend/src/app/features/header/header.component.ts:9]: This should be `{}` per ESLint.

File: frontend/src/sass/_color-palettes.scss [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  100: #aea9ec,
+  200: #8e86e4,
+  300: #6d63dd,
+  400: #5d52d9,
+  500: #544ac3,
+  600: #4a42ae,
+  700: #413998,
+  800: #2f296d,
+  900: #252157,
+  // A100: #7895a2,
```
Comment (bob-fornal) [inline frontend/src/sass/_color-palettes.scss:14]: If the file's not needed, lets remove it.

File: frontend/src/sass/_index.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+@forward 'color-palettes';
```
Comment (bob-fornal) [inline frontend/src/sass/_index.scss:1]: Feel free to remove the files that aren't needed.

File: frontend/src/sass/_themes.scss [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+  200: #544ac3, // For slide toggle,
+  contrast : (
+    main: $light-primary-text,
+    lighter: $dark-primary-text,
+    darker: $light-primary-text,
+  )
+);
+$theme-primary: mat.define-palette($mat-primary, main, lighter, darker);
+
+// body {
```
Comment (bob-fornal) [inline frontend/src/sass/_themes.scss:173]: You can remove this if it's not needed.

File: frontend/src/sass/_themes.scss [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+    warn: $theme-warn,
+    is-dark: false,
+    foreground: $mat-light-theme-foreground,
+    background: $mat-light-theme-background,
+  ),
+  typeography: $fontConfig,
+  density: 0,
+);
+
+// $theme: (
```
Comment (bob-fornal) [inline frontend/src/sass/_themes.scss:249]: These can be removed, rather than commented out.

**Review by bob-fornal — CHANGES_REQUESTED** [review]

File: frontend/src/sass/_themes.scss (hunk 10 above)
Comment (katiehom) [reply frontend/src/sass/_themes.scss]: I removed the commented out the unneeded commented-out code.

File: frontend/src/sass/_themes.scss (hunk 9 above)
Comment (katiehom) [reply frontend/src/sass/_themes.scss]: I removed the unused "secondary" theme.

File: frontend/src/sass/_index.scss (hunk 8 above)
Comment (katiehom) [reply frontend/src/sass/_index.scss]: I removed `_index.scss`

File: frontend/src/sass/_color-palettes.scss (hunk 7 above)
Comment (katiehom) [reply frontend/src/sass/_color-palettes.scss]: I removed `_color-palettes.scss`

File: frontend/src/app/features/header/header.component.ts (hunk 6 above)
Comment (katiehom) [reply frontend/src/app/features/header/header.component.ts]: Fixed

File: frontend/src/app/features/grade-level-selection/grade-level-selection.component.scss (hunk 3 above)
Comment (katiehom) [reply frontend/src/app/features/grade-level-selection/grade-level-selection.component.scss]: You are correct. I added those in.

File: frontend/src/app/features/assistance-selection/assistance-selection.component.html (hunk 2 above)
Comment (katiehom) [reply frontend/src/app/features/assistance-selection/assistance-selection.component.html]: I changed it back to `primary`. We used `accent` to test that it was working.

File: frontend/src/app/app.component.ts (hunk 1 above)
Comment (katiehom) [reply frontend/src/app/app.component.ts]: I have removed this line.

File: frontend/src/app/app.component.scss
Comment (katiehom) [reply frontend/src/app/app.component.scss:1]: I removed `app.component.scss`

File: frontend/src/app/features/grade-level-selection/grade-level-selection.component.ts (hunk 4 above)
Comment (katiehom) [reply frontend/src/app/features/grade-level-selection/grade-level-selection.component.ts]: I added a comma.

**Review by bob-fornal — APPROVED** [review]

**Review by dhess268 — APPROVED** [review]

**Pull request merged**